gm4_liquid_tanks/data/liquid_tanks/functions/level_report_ray.mcfunction

```diff
@@ -0,0 +1,9 @@
+#@s = all players at @s
+#run from main
+
+execute unless entity @s[tag=gm4_lt_found_tank] anchored eyes positioned ^ ^ ^.5 if block ~ ~ ~ hopper align xyz positioned ~0.5 ~ ~0.5 if entity @e[type=armor_stand,tag=gm4_liquid_tank,distance=..0.5] run function #liquid_tanks:level_report_check
```

Comment:
I don't see how `unless entity @s[tag=gm4_lt_found_tank]` can ever do something here

Reply:
If the player is not looking at a liquid tank and thus does not have the tag "gm4_lt_found_tank" then the execute will stop processing the rest of the command, saving on lag.

Reply:
Yes, but the only places this tag is set is in functions called from inside this function and it gets removed at the end of this function. Therefor the tag will never be set when it checks for this on line 4

Reply:
Oh, I see what you mean.

Reply:
You are right in that the first step of the ray will never have that tag set. The rest can though.